### PR TITLE
[MaterializedView] modify ShowMaterializedViewStmt for MaterializedView

### LINKED ISSUE #4784: [MaterializedView] modify ShowMaterializedViewStmt for MaterializedView [P2]
## Feature request

**Is your feature request related to a problem? Please describe.**

to #3680

**Describe the solution you'd like**

should add analyze api in ShowMaterializedViewStmt for new MaterializedView

### PR DESCRIPTION
**Pull request by Crystal-LiuJing**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #4784

## Problem Summary(Required) ：

1 support where or like in ShowMaterializedViewStmt 
2 add analyze api in ShowMaterializedViewStmt for new MaterializedView

### CODE REVIEW

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowMaterializedViewStmt.java [hunk 1]
```diff
@@ -41,10 +46,30 @@ public class ShowMaterializedViewStmt extends ShowStmt {
...
+        this.where = null;
+    }
+
+    public ShowMaterializedViewStmt(String db, String pattern) {
+        this.db = db;
+        this.pattern = pattern;
+        this.where = null;
+    }
+
+    public ShowMaterializedViewStmt(String db, String pattern, Expr where) {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowMaterializedViewStmt.java:69]:
```suggestion
    public ShowMaterializedViewStmt(String db,  Expr where) { this.pattern = null;
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 18 / 133 (13.53%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|69|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowMaterializedViewStmt.java|8|25|32.00%|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|10|10|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 25 / 133 (18.80%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|69|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowMaterializedViewStmt.java|15|25|60.00%|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|10|10|100.00%|

File: gensrc/thrift/FrontendService.thrift [hunk 2]
```diff
@@ -318,12 +318,20 @@ struct TGetTablesParams {
 
 struct TTableStatus {
     1: required string name
-    2: required string type
-    3: required string comment
+    2: optional string type
+    3: optional string comment
```
Comment (ABingHuang) [inline gensrc/thrift/FrontendService.thrift:322]: is there any back-compatible problem here? have you test this change?

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowMaterializedViewTest.java [hunk 3]
```diff
@@ -52,6 +52,29 @@ public void testNormal() throws Exception {
...
+        com.starrocks.sql.analyzer.Analyzer.analyze(stmt, ctx);
+        Assert.assertEquals("bcd", stmt.getPattern());
+        Assert.assertEquals("SHOW MATERIALIZED VIEW FROM testCluster:abc LIKE 'bcd'", stmt.toString());
+        Assert.assertEquals("testCluster:abc", stmt.getDb());
+        Assert.assertEquals(5, stmt.getMetaData().getColumnCount());
+        Assert.assertEquals("id", stmt.getMetaData().getColumn(0).getName());
+        Assert.assertEquals("name", stmt.getMetaData().getColumn(1).getName());
+        Assert.assertEquals("database_name", stmt.getMetaData().getColumn(2).getName());
+        Assert.assertEquals("text", stmt.getMetaData().getColumn(3).getName());
+        Assert.assertEquals("rows", stmt.getMetaData().getColumn(4).getName());
```
Comment (ABingHuang) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowMaterializedViewTest.java:77]: add test case for where

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 4]
```diff
@@ -330,6 +337,92 @@ public TListTableStatusResult listTableStatus(TGetTablesParams params) throws TE
         return result;
     }
 
+    // list MaterializedView table match pattern
+    public void listMaterializedViewStatus(List<TTableStatus> tablesResult, long limit, PatternMatcher matcher,
+                                           UserIdentity currentUser, String dbName) {
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
+        if (db == null) {
+            return;
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:345]: better to add a warning log

File: fe/fe-core/src/main/java/com/starrocks/analysis/SchemaTableType.java [hunk 5]
```diff
@@ -42,6 +42,7 @@ public enum SchemaTableType {
             TSchemaTableType.SCH_GLOBAL_VARIABLES),
     SCH_KEY_COLUMN_USAGE("KEY_COLUMN_USAGE", "KEY_COLUMN_USAGE",
             TSchemaTableType.SCH_KEY_COLUMN_USAGE),
+    SCH_MATERIALIZED_VIEWS("MATERIALIZED_VIEWS","MATERIALIZED_VIEWS",TSchemaTableType.SCH_MATERIALIZED_VIEWS),
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/analysis/SchemaTableType.java:45]:
```suggestion
    SCH_MATERIALIZED_VIEWS("MATERIALIZED_VIEWS", "MATERIALIZED_VIEWS", TSchemaTableType.SCH_MATERIALIZED_VIEWS),
```
add space

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 66 / 149 (44.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|70|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|14|16|87.50%|
|:large_blue_circle:|com/starrocks/analysis/ShowMaterializedViewStmt.java|39|40|97.50%|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|10|10|100.00%|

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Review by HangyuanLiu — DISMISSED** [review]

**Review by Astralidea — APPROVED** [review]

File: gensrc/thrift/FrontendService.thrift [hunk 6]
```diff
@@ -318,12 +318,20 @@ struct TGetTablesParams {
 
 struct TTableStatus {
     1: required string name
-    2: required string type
-    3: required string comment
+    2: optional string type
```
Comment (Astralidea) [inline gensrc/thrift/FrontendService.thrift:321]: If previous is required , for later versions must be require

**Review by Astralidea — CHANGES_REQUESTED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 67 / 155 (43.23%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|74|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Table.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|14|16|87.50%|
|:large_blue_circle:|com/starrocks/analysis/ShowMaterializedViewStmt.java|39|40|97.50%|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|10|10|100.00%|

File: gensrc/thrift/FrontendService.thrift [hunk 7]
```diff
@@ -324,6 +324,14 @@ struct TTableStatus {
     5: optional i64 last_check_time
     6: optional i64 create_time
     20: optional string ddl_sql
+    // id for materialized view
+    21: optional string id
+    // database_name for materialized view
+    22: optional string database_name
```
Comment (kangkaisen) [inline gensrc/thrift/FrontendService.thrift:330]: for information_schema.tables and information_schema.views, thet both needn't database_name in thrift, but could get database from  them, so I think we needn't this field.

File: gensrc/thrift/FrontendService.thrift [hunk 8]
```diff
@@ -324,6 +324,14 @@ struct TTableStatus {
...
+    // ddl_sql for materialized view
+    23: optional string text
```
Comment (kangkaisen) [inline gensrc/thrift/FrontendService.thrift:332]: for the  information_schema.views, we use ddl_sql, I think for  materialized view, we also could use ddl_sql.

File: gensrc/thrift/FrontendService.thrift [hunk 9]
```diff
@@ -324,6 +324,14 @@ struct TTableStatus {
     5: optional i64 last_check_time
     6: optional i64 create_time
     20: optional string ddl_sql
+    // id for materialized view
+    21: optional string id
```
Comment (kangkaisen) [inline gensrc/thrift/FrontendService.thrift:328]: why we need id for  materialized view? for table and view, we only have name, not id.

File: gensrc/thrift/FrontendService.thrift (hunk 7 above)
Comment (Crystal-LiuJing) [reply gensrc/thrift/FrontendService.thrift:330]: the field here is for sql like 'SHOW MATERIALIZED VIEW WHERE 1=1;' , and it is same with the show mv return field

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 67 / 153 (43.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|72|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Table.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|14|16|87.50%|
|:large_blue_circle:|com/starrocks/analysis/ShowMaterializedViewStmt.java|39|40|97.50%|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|10|10|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.3
¡Muchas gracias!

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**